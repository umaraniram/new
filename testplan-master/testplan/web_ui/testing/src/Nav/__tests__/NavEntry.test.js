import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";
import {Badge} from 'reactstrap';

import NavEntry from '../NavEntry';

function defaultProps() {
  return {
    name: 'entry name',
    status: 'passed',
    type: 'testplan',
    caseCountPassed: 0,
    caseCountFailed: 0,
  };
}

describe('NavEntry', () => {
  const props = defaultProps();

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('shallow renders the correct HTML structure', () => {
    const navEntry = shallow(
        <NavEntry {...props} />
    );
    expect(navEntry).toMatchSnapshot();
  });

  it('when prop status="failed" name div and Badge have correct styles',
     () => {
    const failProps = {...props, status: 'failed'}
    const navEntry = shallow(
      <NavEntry {...failProps} />
    );
    expect(navEntry).toMatchSnapshot();
  });

  it('when prop status="xfail" name div and Badge have correct styles',
     () => {
    const failProps = {...props, status: 'xfail'}
    const navEntry = shallow(
      <NavEntry {...failProps} />
    );
    expect(navEntry).toMatchSnapshot();
  });
});
