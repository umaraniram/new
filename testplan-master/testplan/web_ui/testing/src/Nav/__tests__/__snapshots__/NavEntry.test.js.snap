// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavEntry shallow renders the correct HTML structure 1`] = `
<div
  className="d-flex justify-content-between align-items-center"
  style={
    Object {
      "height": "1.5em",
    }
  }
>
  <div
    className="entryName_1dd7qci-o_O-passed_1kh4m98"
    title="entry name"
  >
    entry name
  </div>
  <div
    className="entryIcons_1aztxxt"
  >
    <i
      className="entryIcon_9gb3m5"
      title="passed/failed testcases"
    >
      <span
        className="passed_1kh4m98"
      >
        0
      </span>
      /
      <span
        className="failed_11es5k7"
      >
        0
      </span>
    </i>
    <Badge
      className="entryIcon_9gb3m5-o_O-passedBadge_1isorc2-o_O-badge_1nusemj"
      color="secondary"
      pill={true}
      tag="span"
      title="testplan"
    >
      TP
    </Badge>
  </div>
</div>
`;

exports[`NavEntry when prop status="failed" name div and Badge have correct styles 1`] = `
<div
  className="d-flex justify-content-between align-items-center"
  style={
    Object {
      "height": "1.5em",
    }
  }
>
  <div
    className="entryName_1dd7qci-o_O-failed_11es5k7"
    title="entry name"
  >
    entry name
  </div>
  <div
    className="entryIcons_1aztxxt"
  >
    <i
      className="entryIcon_9gb3m5"
      title="passed/failed testcases"
    >
      <span
        className="passed_1kh4m98"
      >
        0
      </span>
      /
      <span
        className="failed_11es5k7"
      >
        0
      </span>
    </i>
    <Badge
      className="entryIcon_9gb3m5-o_O-failedBadge_1i8mqax-o_O-badge_1nusemj"
      color="secondary"
      pill={true}
      tag="span"
      title="testplan"
    >
      TP
    </Badge>
  </div>
</div>
`;

exports[`NavEntry when prop status="xfail" name div and Badge have correct styles 1`] = `
<div
  className="d-flex justify-content-between align-items-center"
  style={
    Object {
      "height": "1.5em",
    }
  }
>
  <div
    className="entryName_1dd7qci-o_O-unstable_eujaab"
    title="entry name"
  >
    entry name
  </div>
  <div
    className="entryIcons_1aztxxt"
  >
    <i
      className="entryIcon_9gb3m5"
      title="passed/failed testcases"
    >
      <span
        className="passed_1kh4m98"
      >
        0
      </span>
      /
      <span
        className="failed_11es5k7"
      >
        0
      </span>
    </i>
    <Badge
      className="entryIcon_9gb3m5-o_O-unstableBadge_15sg10d-o_O-badge_1nusemj"
      color="secondary"
      pill={true}
      tag="span"
      title="testplan"
    >
      TP
    </Badge>
  </div>
</div>
`;
